### ISSUE TITLE
hive query time booms after using filters

### ISSUE THREAD

**Issue by sandflee** [datafuselabs/databend#7611]
**Summary**
it takes 16s to run the origin sql,  if removing the related filter, it just used 2.2s, 
according to the cpu frame graph, `transform_filter_v2` takes 80%+ cpu. 

```
select sum(gmv_direct)/ 7 as gmv_direct,
       sum(gmv) / 7 as gmv
from online_table.xx
where
p_date between '20210101' and '20210107'
 and resource_type not in ('A', 'B')
;
```

go pprof result:
[pprof.samples.cpu.012.pb.gz](https://github.com/datafuselabs/databend/files/9565360/pprof.samples.cpu.012.pb.gz)

**Comment by sandflee** [comment]
if just using resource_type filter, it takes 15.2s, while just using  p_date filter using  5.5s

**Comment by zhang2014** [comment]
> if just using resource_type filter, it takes 15.2s, while just using p_date filter using 5.5s

Is this because of the bloom filter? cc: @dantengsky

**Comment by dantengsky** [comment]
> > if just using resource_type filter, it takes 15.2s, while just using p_date filter using 5.5s
> 
> Is this because of the bloom filter? cc: @dantengsky

`between` and  `not in ` will not trigger the bloom filter index (although `not in` could benefit from the bloom filter, but currently, only the `=` operator will trigger bloom index) , it looks like there are some other bottlenecks in this case.

**Comment by sandflee** [comment]
`InFunction` costs 60% cpu time

**Comment by BohuTANG** [comment]
It would be the `InFunction` performance issue, cc @sundy-li

**Comment by sundy-li** [comment]
Fixed in #7645
